src/app/body/all-products/filters/filters.component.ts

```diff
@@ -1,15 +1,138 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, ElementRef, OnInit } from '@angular/core';
+import { ProductService } from '../../../services/product.service';
+import { Product } from '../../../models/product.model';
+import { FormArray, FormControl, FormGroup } from '@angular/forms';
+import { Options } from '@angular-slider/ngx-slider';
 
 @Component({
   selector: 'app-filters',
   templateUrl: './filters.component.html',
   styleUrls: ['./filters.component.scss']
 })
 export class FiltersComponent implements OnInit {
+  private products: Product[];
+  public form: FormGroup;
+  public categories: string[];
+  public brands: string[];
+  public rating = Array(5);
+  public sliderOptions: Options;
+  public selectedValue = 'Select';
+  public sortOptions = [
+    {value: 'price', operator: '-', label: 'Price: Low to High'},
+    {value: 'price', operator: '+', label: 'Price: High to Low'},
+    {value: 'rating', operator: '-', label: 'Rating: Low to High'},
+    {value: 'rating', operator: '+', label: 'Rating: High to Low'},
+  ];
 
-  constructor() { }
+  constructor(private productService: ProductService,
+              private element: ElementRef) { }
 
   ngOnInit(): void {
+    this.products = this.productService.products.getValue();
+    this.categories = this.getUniqueCategories();
+    this.brands = this.getBrandsFromProducts();
+    this.form = new FormGroup({
+      categories: new FormControl(''),
+      brands: new FormArray([]),
+      rating: new FormArray([]),
+      price: new FormGroup({
+        minPrice: new FormControl(Math.floor(this.getMinPrice(this.products))),
+        maxPrice: new FormControl(Math.ceil(this.getMaxPrice(this.products)))
+      })
+    });
+    this.sliderOptions = {
+      floor: this.form.controls.price.value.minPrice,
+      ceil: this.form.controls.price.value.maxPrice,
+      step: 1
+    };
```

Comment:
it must be in a separate method or methods.
Please, try keep ngOnInit clen.

Reply:
fixed